src/plain.jl

```diff
@@ -606,6 +606,27 @@ function h5open(f::Function, args...)
     end
 end
 
+if VERSION >= v"0.4"
+    function h5rewrite(f::Function, filename::AbstractString, args...)
+        tmppath,tmpio = mktemp(dirname(filename))
+        close(tmpio)
+
+        try
+            fid = h5open(tmppath, "w", args...)
```

Comment:
the signature of H5freopen is "hid_t H5Freopen(hid_t file_id )".  Does that work with files that were opened with plain old open instead of H5Fopen?